### PR TITLE
Integrate boundary fix from core

### PR DESCRIPTION
**Pull request by OpenSuze**
TODO
- [x] Add better tests
- [x] Update doc
- [x] Merge #168 before

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

Integration test of https://github.com/powsybl/powsybl-core/pull/3169

**What kind of change does this PR introduce?**

Bug fix

**What is the current behavior?**

The total net position might be non zero even if all dangling lines are paired. 

**What is the new behavior (if this is a feature change)?**
If all dangling lines are paired, the total net position is null

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**Other information**:

PyPowSyBl might be broken with this PR

### CODE REVIEW

**Comment by OpenSuze** [comment]
Waiting for PowSyBl Core 6.5.0-RC2

File: flow-decomposition/src/test/java/com/powsybl/flow_decomposition/NetPositionTest.java [hunk 1]
```diff
@@ -122,4 +133,101 @@ void testHvdcLinesNaN() {
...
+        assertEquals(0, tieLine.getDanglingLine1().getBoundary().getP() + tieLine.getDanglingLine2().getBoundary().getP(), DOUBLE_TOLERANCE);
+
+        Map<Country, Double> netPositions = NetPositionComputer.computeNetPositions(network);
+        assertEquals(0, netPositions.values().stream().mapToDouble(Double::doubleValue).sum(), DOUBLE_TOLERANCE);
+        assertEquals(-103.973, netPositions.get(Country.BE), DOUBLE_TOLERANCE);
+        assertEquals(103.973, netPositions.get(Country.FR), DOUBLE_TOLERANCE);
+    }
+
+    @Test
+    void testTieLineWithDisconnectedDanglingLine() {
```
Comment (caioluke) [inline flow-decomposition/src/test/java/com/powsybl/flow_decomposition/NetPositionTest.java:210]: Change test name? I don't see any disconnected dangling lines.

Maybe something like testTieLineWithNonZeroSumOfNetPositions.

File: docs/flow_decomposition/algorithm-description.md [hunk 2]
```diff
@@ -13,7 +13,21 @@ Below is the concrete description of the algorithm implemented in PowSyBl.
...
+
+The sum of net positions is equal to the opposite of the sum of the power on the boundary side of each unpaired half line.
+
+$$\sum_{\text{zone }z} \mathrm{NP}(z) = -\sum_{\text{unpaired half line }h} P_{\text{boundary side}}(h)$$
+
+where:
+- $\mathrm{NP}(\text{zone})$ is the net position of the zone $z$.
+- $P_{\text{boundary side}}(h)$ is the power on the boundary side of the unpaired half line $h$
+
+> **_NOTE:_** If all half lines are merged, the sum of net position is null.
```
Comment (caioluke) [inline docs/flow_decomposition/algorithm-description.md:30]: Missing "s" in net position**s**.

Also, for me, the word `null` makes me think of the Java null.

I would prefere something like

> **_NOTE:_** If all half lines are merged, the sum of net positions is zero.

**Review by caioluke — COMMENTED** [review]
Great job! Ready to merge, just wait for the new powsybl-core release.

**Comment by OpenSuze** [comment]
Thanks for the review @caioluke !

**Review by phiedw — APPROVED** [review]

**Pull request merged**